README.md

```diff
@@ -347,12 +347,39 @@ Julia uses the following external libraries, which are automatically downloaded
 [mbedtls]:      https://tls.mbed.org/
 [pkg-config]:   https://www.freedesktop.org/wiki/Software/pkg-config/
 
+### Notes for distribution package maintainers
+
+Package maintaners will typically want to make use of system libraries where possible. Please refer to the above version requirements and notes below.
+
+Currently community maintained packages are:
+- Fedora: [official](https://src.fedoraproject.org/rpms/julia), [Copr (for backports)](https://copr.fedorainfracloud.org/coprs/nalimilan/julia/)
+- [Arch](https://www.archlinux.org/packages/community/x86_64/julia/)
+- [Gentoo](https://packages.gentoo.org/packages/dev-lang/julia)
+- [Raspbian](https://github.com/JuliaBerry/julia-raspbian)
+
+Additionally, the following unmaintained packages may still be useful:
+- [Homebrew](https://github.com/staticfloat/homebrew-julia)
+- [Ubuntu PPA](https://launchpad.net/~staticfloat/+archive/ubuntu/juliareleases) (0.5)
+- [Debian](https://packages.debian.org/sid/julia) (0.4)
+
 ### System Provided Libraries
 
 If you already have one or more of these packages installed on your system, you can prevent Julia from compiling duplicates of these libraries by passing `USE_SYSTEM_...=1` to `make` or adding the line to `Make.user`. The complete list of possible flags can be found in `Make.inc`.
 
 Please be aware that this procedure is not officially supported, as it introduces additional variability into the installation and versioning of the dependencies, and is recommended only for system package maintainers. Unexpected compile errors may result, as the build system will do no further checking to ensure the proper packages are installed.
 
+### LLVM
+
+The most complicated dependency is LLVM, for which we require version 3.9 with some additional patches (LLVM is not backward compatible). We recommend either:
+ - adding the patches to the LLVM 3.9 package of the distribution (note: these are backported upstream bug fixes, _not_ Julia-specific patches: all have been contributed into upstream LLVM), or
```

Comment:
Maybe add a note the the list of patches is available at `deps/llvm.mk`? 

Maybe https://github.com/JuliaLang/julia/blob/251a501def6a52d497784ad1ca8206579d814bce/deps/llvm.mk#L517-L519 also needs special mentione. E.g. the Julia specific version of `llvm` is modified so that it can be used safely with other versions of `llvm` (4.0 or a system 3.9). This is only true for Julia 0.7+, before that collisions between the Julia version and the system version are likely.

Reply:
Maybe that note would be better if it appeared in the Makefile itself? (it does appear apart from the others)

Reply:
@vchuravy Would you mind adding something to that effect (since I don't quite understand these issues)?

Reply:
What Valentine just said seems to conflict with the text; that sure seems like a julia-specific patch.  (But does it matter if we're using distribution-packaged LLVM?  We don't care about conflicts then  ight?)

Reply:
I'm not sure what this patch does exactly, but we should leave packagers deal with the issue of parallel installation of multiple LLVM versions. It's kind of complex, with some hacking and lots of options to set, and it's far beyond the scope of Julia-specific patches. If that's useful, you can point at how Fedora handles this: https://src.fedoraproject.org/rpms/llvm3.9/blob/master/f/llvm3.9.spec